packages/deployer/src/compiler.ts

```diff
@@ -105,25 +115,43 @@ export class Compiler {
     public async compileAsync(): Promise<void> {
         await createDirIfDoesNotExistAsync(this._artifactsDir);
         await createDirIfDoesNotExistAsync(SOLC_BIN_DIR);
-        this._contractSources = await Compiler._getContractSourcesAsync(this._contractsDir);
-        _.forIn(this._contractSources, this._setContractSpecificSourceData.bind(this));
-        const fileNames = this._specifiedContracts.has(ALL_CONTRACTS_IDENTIFIER)
-            ? _.keys(this._contractSources)
+        this._contractSources = {};
+        for(let contractDir of Array.from(this._contractDirs.values())) {
+            let sources = await Compiler._getContractSourcesAsync(contractDir.path, contractDir.path);
+            _.forIn(sources, (source, sourceFilePath) => {
+                // Construct a unique ID for this source file
+                const sourceFileId:string = constructSourceFileId(contractDir.namespace, sourceFilePath);
```

Comment:
You can skip over most of these types as they are inferred by the return type of the function or the value assigned.
```typescript
const sourceFileId = constructSourceFileId(contractDir.namespace, sourceFilePath);
```